packages/common/src/store/pages/chat/sagas.ts

```diff
@@ -134,6 +182,75 @@ function* doFetchMoreChats() {
   }
 }
 
+/**
+ * Gets all messages newer than what we currently have
+ */
+function* doFetchLatestMessages(
+  action: ReturnType<typeof fetchLatestMessages>
+) {
+  const { chatId } = action.payload
+  try {
+    const audiusSdk = yield* getContext('audiusSdk')
+    const sdk = yield* call(audiusSdk)
+
+    // Update the chat too to keep everything in sync
+    yield* call(doFetchChat, { chatId })
+    const chat = yield* select((state) => getChat(state, chatId))
+    const after = chat?.messagesSummary?.next_cursor
+
+    let hasMoreUnread = true
+    let data: ChatMessage[] = []
+    let before: string | undefined
+    let firstResponse: TypedCommsResponse<ChatMessage[]> | undefined
+    while (hasMoreUnread) {
+      // This will get all messages sent after what we currently got, starting at the most recent,
+      // and batching by MESSAGE_PAGE_SIZE. Sends one extra request to get the 0 response but oh well
+      const response = yield* call([sdk.chats, sdk.chats.getMessages], {
+        chatId,
+        before,
+        after: chat?.messagesSummary?.next_cursor,
+        limit: MESSAGES_PAGE_SIZE
+      })
+      data = data.concat(response.data)
+      hasMoreUnread =
+        // If we have no more messages with our after filter, we're done
+        (!!after && response.data.length > 0) ||
+        // If we don't have an after filter, eg on first messages load, then go until we get all the unreads
+        (!after &&
+          !!chat?.unread_message_count &&
+          chat.unread_message_count >
+            (response.summary?.next_count ?? 0) + data.length)
+      before = response.summary?.prev_cursor
```

Comment:
Very possibly not getting this because it's late, but what happens here with the inclusion of both a before and after cursor in the request?

Reply:
It only fetches messages between before/after